### PR TITLE
HURUMap Source

### LINKED ISSUE #775: @hurumap/core/Source

### PR DESCRIPTION
**Pull request by kelvinkipruto**
## Description
This PR Moves the HURUMap `Source` Component as a reusable component. It also creates a new package `@hurumap/next` to handle apps with Next.JS dependencies.

Fixes #775

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Screenshots
image

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/climatemappedafrica/src/components/HURUmap/KeyMetric/index.js [hunk 1]
```diff
@@ -67,7 +66,12 @@ function KeyMetric({
           {parentValue}
         </Typography>
       )}
-      <Source href={url} classes={{ root: classes.source }}>
+      <Source
+        href={url}
+        sx={(theme) => ({
+          margin: `${theme.typography.pxToRem(20)} 0`,
```
Comment (kilemensi) [inline apps/climatemappedafrica/src/components/HURUmap/KeyMetric/index.js:72]: The useStyles you're replacing says `margin-top: 10`
```suggestion
          marginTop: typography.pxToRem(10),
```

File: apps/pesayetu/src/components/HURUmap/KeyMetric/index.js [hunk 2]
```diff
@@ -67,7 +66,12 @@ function KeyMetric({
           {parentValue}
         </Typography>
       )}
-      <Source href={url} classes={{ root: classes.source }}>
+      <Source
+        href={url}
+        sx={(theme) => ({
+          margin: `${theme.typography.pxToRem(20)} 0`,
+        })}
```
Comment (kilemensi) [inline apps/pesayetu/src/components/HURUmap/KeyMetric/index.js:73]:
```suggestion
        sx={(theme) => ({
          marginTop: theme.typography.pxToRem(10),
        })}
```

File: packages/hurumap-core/package.json [hunk 3]
```diff
@@ -67,6 +67,7 @@
     "react-leaflet": "^4.2.1"
   },
   "dependencies": {
-    "prop-types": "^15.8.1"
+    "prop-types": "^15.8.1",
+    "@commons-ui/next": "workspace:*"
```
Comment (kilemensi) [inline packages/hurumap-core/package.json:71]: `@commons-ui/next` **shouldn't** be used in `@hurumap/core`

File: packages/hurumap-next/.eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  root: true,
+  extends: ["eslint-config-commons-ui"],
```
Comment (kilemensi) [inline packages/hurumap-next/.eslintrc.js:3]:
```suggestion
  extends: ["eslint-config-commons-ui/next"],
```

File: packages/hurumap-next/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# @commons-ui/next
```
Comment (kilemensi) [inline packages/hurumap-next/CHANGELOG.md:1]:
```suggestion
# @hurumap/next
```

File: packages/hurumap-next/CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  - @commons-ui/core@2.0.1
+
+## 2.0.0
+
+### Patch Changes
+
+- [#241](https://github.com/CodeForAfrica/ui/pull/241) [`f792694`](https://github.com/CodeForAfrica/ui/commit/f7926944fb505adaa8905dc8aeba04913f56ea85) Thanks [@kilemensi](https://github.com/kilemensi)! - Improved RichTypography to better handle color and relative links
+
+- Updated dependencies [[`f792694`](https://github.com/CodeForAfrica/ui/commit/f7926944fb505adaa8905dc8aeba04913f56ea85)]:
+  - @commons-ui/core@2.0.0
```
Comment (kilemensi) [inline packages/hurumap-next/CHANGELOG.md:19]: Don't think any of this is related to the new package.

```suggestion
```

File: packages/hurumap-next/package.json [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    "@commons-ui/core": "workspace:*",
+    "@mui/material": "^5.14.0",
+    "clsx": "^2.0.0",
+    "next": "~13.4.11",
+    "prop-types": "^15.8.1",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0"
+  },
+  "dependencies": {
+    "clsx": "^2.1.1",
```
Comment (kilemensi) [inline packages/hurumap-next/package.json:63]:
```suggestion
```
We are not working with `useStyles` so not sure we'll ever need `clsx`.

File: packages/hurumap-next/src/Source/Source.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+import { Typography } from "@mui/material";
+import React from "react";
+
+const Source = React.forwardRef(function Source(
+  {
+    children,
+    href,
+    title = "Source",
+    TitleTypographyProps,
+    LinkProps,
```
Comment (kilemensi) [inline packages/hurumap-next/src/Source/Source.js:11]:
```suggestion
    LinkProps,
    TitleTypographyProps,
    children,
    href,
    title = "Source",
```

File: packages/hurumap-next/package.json [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    "react-dom": "^18.3.1",
+    "require-from-string": "^2.0.2",
+    "typescript": "^5.5.3",
+    "webpack": "^5.93.0"
+  },
+  "peerDependencies": {
+    "@babel/core": "^7.22.9",
+    "@commons-ui/core": "workspace:*",
+    "@mui/material": "^5.14.0",
+    "clsx": "^2.0.0",
```
Comment (kilemensi) [inline packages/hurumap-next/package.json:56]:
```suggestion
```

**Review by kilemensi — COMMENTED** [review]
👍🏽

**Review by koechkevin — APPROVED** [review]

**Review by m453h — APPROVED** [review]

**Review by kilemensi — APPROVED** [review]

**Pull request merged**